### DISCUSSION TITLE
nsexec cannot view the processes in the target container

### DISCUSSION THREAD

**Discussion by zhuazi** [chaos-mesh/chaos-mesh#3723] [Q&A] [answered]
## Bug Report

**What version of Kubernetes are you using?**

**What version of Chaos Mesh are you using?**
v2.1.3

**What did you do? / Minimal Reproducible Example**

use nsexec：
![image](https://user-images.githubusercontent.com/4626297/197444899-ee6d5e0a-9d0d-49ac-91e4-4180ce135f23.png)
use nsenter：
![image](https://user-images.githubusercontent.com/4626297/197444953-b0928340-de8a-4869-9fe8-ae1f0623c2c4.png)

**What did you expect to see?**
all the process in the container
**What did you see instead?**

**Output of chaosctl**

**Comment by STRRL** [comment, marked as answer]
have you tried `nsexec` with `ps -ef`?

**Comment by STRRL** [comment]
It would also help a lot if you could tell us what the exactly what you want to do :)

**Comment by zhuazi** [comment]
> hi，I want to query the program pid and program cmd through the specified string.I think it may be the problem with the exec. command method,not the nsexec.

i want to get the processid and processname with the give

**Reply by STRRL** [reply]
you want to get `pid` and name of the process, with what?
